src/Concerns/Trackable.php

```diff
@@ -27,6 +38,11 @@ public function __construct()
         ]);
     }
 
+    private function isUniqueJobAndCanAcquireLock(): bool
+    {
+        return (new UniqueLock(Container::getInstance()->make(Cache::class)))->acquire($this);
```

Comment:
I think this won't work as laravel will acquire this lock before when dispatching this job 🤔